### PR TITLE
fix(Button): add danger variant assistive text

### LINKED ISSUE #8220: Button doesn't communicate danger state to screen reader users
image

If the danger state is important visual information that calls warning to potentially destructive or irreversible processes/actions we need to communicate that to our screen reader users as well. 

This can be done a few ways, but off the top of my head I might try adding a visually hidden `span` with a unique id that can be referenced in an `aria-describedby` attribute on the button itself something like this perhaps

image

### PR DESCRIPTION
**Pull request by dakahn**
Closes #8220 

Adds assistiveText for 'danger' button variant 

#### Testing / Reviewing

- Make sure that `danger` button doesn't throw any Accessibility Checker errors
- With NVDA, JAWS, and VoiceOver ensure that "Danger" is read along with the button's label

### CODE REVIEW

File: packages/react/src/components/Button/Button.js [hunk 1]
```diff
@@ -143,19 +143,33 @@ const Button = React.forwardRef(function Button(
   let otherProps = {
     disabled,
     type,
+    'aria-describedby': kind === 'danger' ? 'danger-description' : null,
```
Comment (tw15egan) [inline packages/react/src/components/Button/Button.js]: I think we need to check if `kind` is `danger`, `danger--tertiary`, or `danger--ghost`. During testing, I only get the `danger` warning for the first button in the `Danger Button` story

File: packages/react/src/components/Button/Button.js [hunk 2]
```diff
@@ -143,19 +143,33 @@ const Button = React.forwardRef(function Button(
...
+        ref={tooltipRef}
+        onMouseEnter={handleMouseEnter}
+        className={`${prefix}--assistive-text`}>
+        {iconDescription}
+      </div>
+    );
+  } else if (kind === 'danger') {
+    assistiveText = (
+      <span id="danger-description" className={`${prefix}--visually-hidden`}>
+        danger
```
Comment (joshblack) [inline packages/react/src/components/Button/Button.js]: Would this need to be configurable / translatable?

Comment (dakahn) [reply packages/react/src/components/Button/Button.js]: allowing it to be configurable via props (and thus translatable) is smart!

**Review by tw15egan — APPROVED** [review]
Voiceover correctly alerting `danger` on danger button variants 

LGTM 👍 ✅

File: packages/elements/src/__tests__/__snapshots__/PublicAPI-test.js.snap [hunk 3]
```diff
@@ -250,7 +250,6 @@ Array [
   "spacing10",
   "spacing11",
   "spacing12",
-  "spacing13",
```
Comment (joshblack) [inline packages/elements/src/__tests__/__snapshots__/PublicAPI-test.js.snap]: Might need to re-build and sync up this snapshot again 👀

**Review by joshblack — APPROVED** [review]

**Pull request merged**